src/ahbicht/validation/validation.py

```diff
@@ -0,0 +1,351 @@
+    soll_is_required=True,
+) -> DataElementValidationResult:
+    """
+    Validates a freetext dataelement, e.g. 'Dokumentennummer'.
+    :param dataelement: the dataelement that should be validated
+    :param segment_requirement: the requirement of the dataelement's segment, e.g. IS_REQUIRED
+    :param soll_is_required: true (default) if SOLL should be handled like MUSS,
+                             false if it should be handled like KANN
+    :return: Validation Result of the Dataelement
+    """
+
+    expression_tree = parse_ahb_expression_to_single_requirement_indicator_expressions(dataelement.ahb_expression)
+    evaluation_result = await evaluate_ahb_expression_tree(expression_tree, entered_input=dataelement.entered_input)
+
+    # requirement constraints
+    requirement_validation_without_input_without_hierachry = map_requirement_validation_values(
+        evaluation_result.requirement_constraint_evaluation_result.requirement_constraints_fulfilled,
+        evaluation_result.requirement_indicator,
+        soll_is_required,
+    )
+    requirement_validation_without_input = combine_requirements_of_different_levels(
+        segment_requirement, requirement_validation_without_input_without_hierachry
+    )
+
+    # format constraints
+    if evaluation_result.format_constraint_evaluation_result.format_constraints_fulfilled:
+        format_validation = FormatValidationValue.FORMAT_CONSTRAINTS_ARE_FULFILLED
+    else:
+        format_validation = FormatValidationValue.FORMAT_CONSTRAINTS_ARE_NOT_FULFILLED
+
+    if dataelement.entered_input:
+        requirement_validation = RequirementValidationValue(str(requirement_validation_without_input) + "_AND_FILLED")
+    else:
+        requirement_validation = RequirementValidationValue(str(requirement_validation_without_input) + "_AND_EMPTY")
+
+    return DataElementValidationResult(
+        requirement_validation=requirement_validation,
+        format_validation=format_validation,
+        format_error_message=evaluation_result.format_constraint_evaluation_result.error_message,
+        hints=evaluation_result.requirement_constraint_evaluation_result.hints,
+    )
+
+
+async def validate_dataelement_valuepool(
+    dataelement: DataElementValuePool,
+    segment_requirement: RequirementValidationValue,
+) -> DataElementValidationResult:
+    """
+    Validates a value pool data element which depends on the requirement status of its segment.
+    :param dataelement: the dataelement that should be validated
+    :param segment_requirement: the requirement of the dataelement's segment, e.g. IS_REQUIRED
+    :returns: Validation Result of the Dataelement
+    """
+    possible_values = []
+
+    # Since the conditions behind the qualifiers only say if this qualifier is possible or not,
+    # the overall requirement is inherited from the dataelement's segment.
+    # Only exception is if no dataelement is possible, see below
```

Comment:
das verstehe ich nicht: wie ist kann ein "data element" not possible sein?

Reply:
Korrekt, da soll qualifier stehen
3c2b0dc0c671f09dd2ec6b756dcc1c0c3cdf2cec